src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1978,6 +2016,30 @@ FixedwingPositionControl::control_auto_landing(const hrt_abstime &now, const flo
 					   false,
 					   radians(_param_fw_p_lim_min.get()));
 	}
+
+	/* Copy thrust and pitch values from tecs */
+	if (_landed) {
+		// when we are landed state we want the motor to spin at idle speed
+		_att_sp.thrust_body[0] = min(_param_fw_thr_idle.get(), 1.f);
+
+	} else {
+		_att_sp.thrust_body[0] = get_tecs_thrust();
+	}
+
+	// decide when to use pitch setpoint from TECS because in some cases pitch
+	// setpoint is generated by other means
+	bool use_tecs_pitch = true;
+
+	// flaring during landing
+	use_tecs_pitch &= !(_land_noreturn_vertical);
+
+	if (use_tecs_pitch) {
```

Comment:
I am personally fine with this change(Since I really don't understand the landing logic in detail), but this will change the logic from
```
if (!_land_no_return_vertical) {
  _att_sp.pitch_body = get_tecs_pitch();
}
```
to
```
if ((_current_altitude < terrain_alt + _landingslope.flare_relative_alt()) &&
	    !_land_noreturn_vertical) {
  _att_sp.pitch_body = get_tecs_pitch();
}
```

As far as I understand. I have applied this in the review commits

Reply:
It is still outside the condition in your most recent commit:
https://github.com/PX4/PX4-Autopilot/blob/7c90ad7143cf24d078735c620c424966af086ee7/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp#L2002-L2004

You are correct that the condition technically changes to also consider the height relative to the flare curve, but if you look through the logic, `land_noreturn_vertical` is only ever set one in that flare logic condition, and in fact is *always* set in that condition, and then "never returns".

I would still vote for putting the `_att_sp.pitch_body = get_tecs_pitch();` inside the `else` condition with the rest of the glideslope logic, since the flare logic above it commands all the way down to direct pitch setpoints `_flare_pitch_sp`, and it just seems error prone to me that we have another check for the `land_noreturn_vertical` condition outside of that main `if` tree. -- in general this whole thing needs to be cleaned up, refactored, etc.. but for now at least we can group things in their respective places before encapsulating into clearer defined units.

Reply:
Done!